scripts/build/cardano-sl.sh

```diff
@@ -327,19 +327,25 @@ echo "'explorer' flag: $explorer"
 
 for prj in $to_build; do
 
+  if [[ $prj == "cardano-sl-core" ]]; then
+    ghc_opts_2="$ghc_opts -DGITREV=`git rev-parse HEAD`"
+  else
+    ghc_opts_2="$ghc_opts"
+  fi
+
   echo -e "Building $prj\n"
-  sbuild="stack build --ghc-options=\"$ghc_opts\" $commonargs $norun --dependencies-only $args $prj"
+  sbuild="stack build --ghc-options=\"$ghc_opts_2\" $commonargs $norun --dependencies-only $args $prj"
   echo -e "$sbuild\n"
   eval $sbuild
 
   if [[ $no_code == true ]]; then
-    ghc_opts_2="$ghc_opts -fwrite-interface -fno-code"
+    ghc_opts_3="$ghc_opts2 -fwrite-interface -fno-code"
   else
-    ghc_opts_2="$ghc_opts"
+    ghc_opts_3="$ghc_opts2"
```

Comment:
Why did you switch to `ght_opts_N`? I personally find it less convenient than overriding `ghc_opts`.